lib/AST/Module.cpp

```diff
@@ -674,7 +675,8 @@ void ModuleDecl::getTopLevelDeclsWhereAttributesMatch(
 }
 
 void SourceFile::getTopLevelDecls(SmallVectorImpl<Decl*> &Results) const {
-  Results.append(Decls.begin(), Decls.end());
+  auto decls = getTopLevelDecls();
+  Results.append(decls.begin(), decls.end());
```

Comment:
The calltree for this is pretty big, but are any of the callers actually mutating this out parameter? If not, let's delete this entrypoint and migrate them all.

Reply:
@CodaFi It's mainly used to override `FileUnit`'s `getTopLevelDecls`, which uses an out-parameter. We'd need to add caching for the top level decls of a `ClangModuleUnit` and `SerializedASTFile` to switch over to returning an `ArrayRef`, though maybe that's worth it?